### PR TITLE
virttest.utils_test.qemu: move migration codes to a single file

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei xuwei@redhat.com

### CODE REVIEW

**Comment by XueqiangWei** [comment]
ID 1342353

File: virttest/utils_test/qemu/__init__.py [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+
+def pin_vm_threads(vm, node):
+    """
+    Pin VM threads to single cpu of a numa node
+
+    :param vm: VM object
+    :param node: NumaNode object
+    """
+    if len(vm.vcpu_threads) + len(vm.vhost_threads) < len(node.cpus):
+        for i in vm.vcpu_threads:
```
Comment (xutian) [inline virttest/utils_test/qemu/__init__.py]: sorry, see it.

File: virttest/utils_test/qemu/migration.py [hunk 2]
```diff
@@ -1377,468 +1211,459 @@ def mig_wrapper(vm, cancel_delay, dsthost, vm_ports,
...
+
+        """
+        install stressapptest
+        """
+
+        vm = self.env.get_vm(self.params["main_vm"])
+        session = vm.wait_for_login(timeout=self.login_timeout)
+        app_repo = "git clone https://github.com/stressapptest/" \
+                   "stressapptest.git"
+        stressapptest_insatll_cmd = "rm -rf stressapptest " \
```
Comment (xutian) [inline virttest/utils_test/qemu/migration.py]: better to use '%s' instead  '+'.

File: virttest/utils_test/qemu/__init__.py [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+from virttest import qemu_monitor
+from virttest.qemu_devices import qdevices
+from virttest.staging import utils_memory
+
+
+def guest_active(vm):
+    o = vm.monitor.info("status")
+    if isinstance(o, str):
+        return "status: running" in o
+    else:
```
Comment (XueqiangWei) [inline virttest/utils_test/qemu/__init__.py]: This is raw code in the framework, I have not changed. And I think it is ok, it judges the return message from hmp or qmp.

File: virttest/utils_test/qemu/__init__.py [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+      lower-level modules (e.g. utils_misc, qemu_vm, aexpect).
+    - Functions in this module should not be used by lower-level modules.
+    - Functions in this module should be used in the right context.
+      For example, a function should not be used where it may display
+      misleading or inaccurate info or debug messages.
+
+:copyright: 2008-2013 Red Hat Inc.
+"""
+
+import logging
```
Comment (XueqiangWei) [inline virttest/utils_test/qemu/__init__.py]: will update, thx

File: virttest/utils_test/qemu/migration.py (hunk 2 above)
Comment (XueqiangWei) [inline virttest/utils_test/qemu/migration.py]: ok, thx

File: virttest/utils_test/qemu/migration.py [hunk 5]
```diff
@@ -1377,468 +1211,459 @@ def mig_wrapper(vm, cancel_delay, dsthost, vm_ports,
...
+            args=(self.test, self.params, self.env,),
+            kwargs={"sub_type": self.bg_stress_test})
+        bg.start()
+        time.sleep(self.WAIT_SHORT)
+
+        def check_running():
+            return session.cmd_status(self.check_running_cmd) == 0
+
+        if self.check_running_cmd:
+            if not utils_misc.wait_for(check_running, timeout=360):
```
Comment (suqinhuang) [inline virttest/utils_test/qemu/migration.py]: suggest to add step description

File: virttest/utils_test/qemu/migration.py [hunk 6]
```diff
@@ -1377,468 +1211,459 @@ def mig_wrapper(vm, cancel_delay, dsthost, vm_ports,
...
+                         "no need to kill.")
+        else:
+            try:
+                s, o = session.cmd_status_output(kill_bg_cmd)
+                logging.info("The output after run kill command: %r" % o)
+                if "No such process" in o or "not found" in o \
+                        or "no running instance" in o:
+                    if session.cmd_status(self.check_running_cmd) != 0:
+                        logging.info("The background test in guest is "
+                                     "finished before kill it.")
```
Comment (suqinhuang) [inline virttest/utils_test/qemu/migration.py]: why do we need this check again, seems it's the same as line 1599

Comment (XueqiangWei) [inline virttest/utils_test/qemu/migration.py]: sometimes, the background test is finished when try to kill it. It seldom happened, in order to handle it, check background test again.

**Comment by xutian** [comment]
Merged, tested and it works for me, thanks @XueqiangWei

**Pull request merged**